common/djangoapps/student/roles.py

```diff
@@ -218,6 +218,33 @@ def users_with_role(self):
         )
         return entries
 
+    def update_org_role(self, user, orgs):
```

Comment:
This method seems out of place.

An instance of `RoleBase` is meant to encapsulate a role name, an optional organization, and an optional course. The operations available on a `RoleBase` instance are adding users, removing users, and checking whether a user has the role.

This method does something completely else. It assumes that the `org` and `course_key` are None, and then mutates the roles across a range of organizations. This creates some weird behaviors. For example, if you executed this:
```python
OrgContentCreatorRole(org='org1').update_org_role(user, ['org2'])`
```
then the user would be assigned to just `org2`.

I recommend changing `update_org_role` in in `auth.py` to something like:
```python
def update_user_role_organizations(caller, role_class, user, *orgs):
    """
    ... docstring
    """
    _check_caller_authority(caller, role)
    for org in orgs:
        role_class(org=org).add_user(user)
```
letting the existing `RoleBase` functions take care of the CourseAccessRole table and caching concerns.